src/components/forms/TextInput.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { throttle, debounce } from 'throttle-debounce';
+import IconInput from 'components/forms/IconInput';
+import DecoratedInput from 'components/forms/DecoratedInput';
+
+export const throttleValue = 500;
+
+const TextInput = ({
+  label,
+  errorText,
+  throttleOnChange,
+  required,
+  onChange,
+  type,
+  IconLeft,
+  IconRight,
+  after,
+  before,
+  ...props
+}) => {
+  // Refer to the following for an explanation on
+  // the throttling / debouncing approach:
+  // https://www.peterbe.com/plog/how-to-throttle-and-debounce-an-autocomplete-input-in-react
```

Comment:
I highly highly recommend reading this article, just to make sure we're aligned on the approach. There may be different approaches / articles we can look into as well.

There's actually another thing in the article that I might steal... throttling if the length of the characters is less than 5, AND if the ending character ever ends in a space. Article explains why you'd want to do that.